docs/articles/documentation/test-api/assertions/index.md

```diff
@@ -0,0 +1,108 @@
+See [Assertion API](assertion-api.md).
+
+This topic contains the following sections.
+
+* [Assertion Structure](#assertion-structure)
+* [Smart Assertion Query Mechanism](#smart-assertion-query-mechanism)
+* [Assertion options](#assertion-options)
+  * [options.timeout](#optionstimeout)
+
+## Assertion Structure
+
+TestCafe assertions start with `expect` method exposed by [test controller](../test-code-structure.html#test-controller)
+which accepts `actual` value argument. Then the assertion method follows. An assertion method accepts expected a value
+and optionally other arguments. E.g. deep equality assertion:
+
+```text
+await t.expect( actual ).eql( expected, message, options );
+```
+
+The sample below demonstrates how to use assertions:
+
+```js
+import { Selector } from 'testcafe';
+
+
+fixture `Example page`
+   .page `http://devexpress.github.io/testcafe/example/`;
+
+
+test('Check property of element', async t => {
+   const developerNameInput = Selector('#developer-name');
+
+   await t
+       .expect(developerNameInput.value).eql('', 'input is empty')
+       .typeText(developerNameInput, 'Peter')
+       .expect(developerNameInput.value).eql('Peter', 'input has text');
+});
+```
+
+## Smart Assertion Query Mechanism
+
+In synchronous functional testing, you can perform the required assertions immediately after test action was executed.
+
+[first picture]
+
+In web, functional tests are asynchronous. This means that we can not get the expected changes immediately after an end-user action.
+For example, the tested web page can send a request to the server for the required data, and this can take some time;
+or an end-user action launches animation after which the web page will reach its final state.
+All of these time gaps can not be pre-calculated, because they depend on various factors: computer performance,
+network connection speed, etc. In this case, if we perform assertions immediately after the test action finished,
+we can get an indefinite result.
+
+[second picture]
+
+To perform asynchronous functional tests, the additional timeout as usually added.
+
+[third picture]
+
+To make such tests stable, you need to add a timeout that will guarantee you a successful completion of required changes.
+Adding such timeouts can increase the test running time because of extra waiting.
+
+The main feature of TestCafe assertions is the use of built-in assertion query mechanism.
+To watch the state of web page elements, TestCafe provides [Selectors](../selecting-page-elements/selectors.md).
+If the TestCafe assertion receives a [Selector's property](../selecting-page-elements/selectors.md#snapshot-api-shorthands)
+as an actual value, TestCafe uses the smart assertion query mechanism:
+if an assertion did not passed, the test does not fail immediately. The assertion retries to pass multiple times and
+each time it re-requests the actual shorthand value. The test fails if the assertion could not complete successfully
+within a timeout.
```

Comment:
**Example.**
Let's create a test for the following web page.
```html
<div id="btn"></div>
<script>
var btn = document.getElementById('btn');

btn.addEventListener(function() { 
    window.setTimeout(function() {
        btn.innerText = 'Loading...';
    }, 100);
});
</script> 
```
The test code will look as follows.

```js
test('Button click', async t => {
    const btn = Selector('#btn');

    await t
        .click(btn)
         // NOTE: Regular assertion will fail immediately, but TestCafe retries to run DOM state
         // assertions many times until this assertion pass successfully within the specified timeout.
        .expect(btn.textContent).contains('Loading...', {timeout:1000}); 
});
```

[Kazakov: pay attention to the specified timeout, that makes the example more clear]